### ISSUE TITLE
UnobservedTaskException's from ReadAheadWithZeroByteReadAsync

### ISSUE THREAD

**Issue by davidni** [dotnet/runtime#75355]
### Description

We are seeing a large number of UnobservedTaskException's as below:

```
System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host..)
 ---> System.IO.IOException: Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host..
 ---> System.Net.Sockets.SocketException (10054): An existing connection was forcibly closed by the remote host.
   --- End of inner exception stack trace ---
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](TIOAdapter adapter, Memory`1 buffer)
   at System.Net.Http.HttpConnection.<CheckUsabilityOnScavenge>g__ReadAheadWithZeroByteReadAsync|44_0()
   --- End of inner exception stack trace ---
```

The fault seems to originate from here, where indeed exceptions will go unhandled:
https://github.com/dotnet/runtime/blob/071fe7ce868a63113143b949c3c4185f9e5578cf/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs#L221-L234

### Reproduction Steps

Unclear but shouldn't be hard to come up with per description above.

### Expected behavior

No UnobservedTaskException's should occur.

### Actual behavior

UnobservedTaskException's are being emitted.

### Known Workarounds

Perhaps this was already fixed here? https://github.com/dotnet/runtime/pull/61263

### Configuration

.NET 6.0.8 on Windows x64

### Other information

Submitting even if it's a dupe so that other can find it easily

**Comment by MihaZupan** [comment]
Duplicate of #61256

> Perhaps this was already fixed here? https://github.com/dotnet/runtime/pull/61263

Sadly not, that change never went in.

**Comment by MihaZupan** [comment]
I'll update the original issue with the stack trace so it's easier to find
